### ISSUE TITLE
Video tutorials

### ISSUE THREAD

**Issue by NathanLovato** [GDquest/Blender-power-sequencer#117]
The goal of this task is to produce long tutorial videos that go beyond just using the add-on: there's still little material to get people up to speed with Blender's video sequencer besides Mickeycal's videos. There's room to offer a different teaching style.

## Suggestions

I'd recommend going for long video tutorials that cover the basics of Blender's VSE and go past that as well. At the end of the video/series the viewers should:

1. Be able to create their own videos from start to finish
2. Understand how Blender's VSE works, if it's a good fit for them, and what its limitations are, especially compared to regular video editing programs

## Idea

The more attractive the videos, the better. I'm hoping to bring some more attention to the VSE with this, and hopefully to attract some more contributors to the motion design/video part of Blender as well. Be it for this add-on or not.

I've tried putting up feature-specific videos but it's more work to set each of them up on Youtube for one, and they don't rank in the search engines thus get little attention. Instead I'm vouching for longer tutorials, up to 30-40 minutes long a piece, well-produced, and always project-driven. 

We could provide the assets/files to get people started and for them to follow along if they want. I'm shooting a little bit around Nara and places I go to, and I'll be recording footage during the cherry blossom season starting next week. I'm looking to open source all that, share it via torrent so the viewers can have decent raw footage to follow along and work with.

## Topics to cover

WIP list. Please feel free to edit and/or comment to suggest more topics:

- Intro to the series: what people will learn, how to take the most out of it, chapters etc.
- Intro to Blender for video editing: strengths and limitations
  - I made and gave a [ presentation ](https://github.com/GDquest/blender-sequencer-presentation) on the matter, can translate it to ENG to save some work
- VSE interface
- Installing both PS and VSE Transform tools add-ons, updates
- Handling footage: import, framerate, transcoding, proxies
- Basic editing: strips, handles, move, cut, trim, slide
- Rendering/exporting
- More strip manipulations: extend, select cuts, 
- Working with pictures: transform strips, using the VSE transform tools
- How to do text: Show in-Blender and external pics as options
- Transitions: fades, crossfades,
- Simple motion design: image animation, scene strips, custom cross transitions with image strips
- Subtitling: texts strips and showing your add-on @doakey3?

*Planning it for 1.1 as 1.0 is almost ready already, and it'd be nice to cover the latest features in the tutorials.*

What do you think @doakey3 @davcri ?

**Comment by MekaClaude** [comment]
Great idea !

**Comment by doakey3** [comment]
I've written up a [rough draft](https://github.com/GDquest/Blender-power-sequencer-docs/tree/master/tutorial_series) for how we might split these up into videos. Let's collaborate and decide exactly how we want this to be. Feel free to edit the rough draft, adding or removing as you see fit.

**Comment by NathanLovato** [comment]
Wow, a lot of details in there. Great work, but note moving forward you don't have to write down all the steps with so many details: it'll be hard to follow stricly once you create the real tutorials, especially after you produced the projects. For example:

```
Demonstrate that shuttling through the clips is laggy
Explain what a proxy video file is and why it fixes the lagginess. Final rendering will use the original source file, not the proxy file.
Generate 25% proxies for all the clips and set the proxy render size accordingly.
Demonstrate how the proxy has sped up shuttling
```

Instead I'd write

```
importing footage b3d & ps way
proxies
```

It's only the 2 of us who are going to work on the tuts. We both have enough experience to get these right, I've seen your work and definitely trust your skills in there.

About the styleguide we're mostly good.

```
If you want to be awesome, finish the video with a quick summary of what was learned
```

We can have some kind of "cheat-sheet" at the end. As long as it doesn't paraphrase the intro it's sensible.

```
a tutorial should be 10-20 minutes or less
```

I disagree on the idea that this is a characteristic of a good tutorial. Sections of the videos can be short so people can come back to them later and/or jump to different chapters but I wouldn't have use constrain the length of the videos because of the average person's  attention span. On Youtube it's probably closer to 1 minute than 10 as well. 

The way you present the video (or give a lecture) is a lot more important when it comes to retention and keeping people's attention. That's why I'd have us focus solely on making a captivating video that'll appeal to people who are interested in making videos.

Which leads me to another point: the target audience. You can't make a video that will work for everyone. Lazy people and those with a short attention span aren't your target when you're covering a program that's as demanding as Blender.

I'm aiming for videos that are both accessible to beginners and paced well-enough for intermediate blender users to enjoy. Blender being Blender, it's not a good pick for everyone, especially not people who aren't comfortable with a computer etc. Looking at your draft steps it's all good for me.

## Commenting on the draft steps

```
(Good luck convincing your friends to buy After Effects so they can help you edit.)
```

Good luck convincing them to use Blender too. There's so much piracy with larger programs this argument wouldn't convince me, even as someone who works in FOSS.

![image](https://user-images.githubusercontent.com/12694995/37862053-2dcd7b9a-2f8a-11e8-8d0a-2b9c7d3122b9.png)

It's beyond the scope of `Intro to Blender for Video Editing`

```
Whatever else from GDQuest's presentation
```

The pres focuses on how Blender can be a good pick for people to edit videos and do motion design. It tries to show that it's an all-in-one package to achieve your creative vision, that there are all the tools you need to do professional work with it... provided that you're willing to invest enough time.

For the rest of the notes: everything holds in one video length-wise. Maybe not theme-wise at first glance but how about creating a **Complete introduction to video editing with Blender** ? Because:

1. The purpose is clear, and so is the skills people will develop beyond learning this or that feature
1. Thus it'll attract people
1. Series/splitting videos *always* leads to a loss in retention and to issues that arise from people starting at video 3 and asking questions that were answered in video 1 or 2 (I get questions and messages pretty much daily from that)
1. It's only one video to manage on Youtube so less work (not only setup and SEO, but community management etc.)

There's more arguments for this like people have only 1 link to care about, releasing multiple videos at once is very bad for their visibility and so is spreading them in time for tutorial series.

As far as the flow is concerned I def. agree about showing how some things work in vanilla Blender before using any add-on.

Great work on the draft anyway, it's good to start with. I'd sprinkle a few more things along the way like the technical constraints you face with Blender, and a tiny bit of troubleshooting as it's easy to get stuck.
After that we can always think about a part 2.

That said gotta go it's closing time at the co-working space!

**Comment by doakey3** [comment]
I really like Andrew Price's [Subway Tutorial](https://www.youtube.com/watch?v=nb6rSMAooDs). I'm thinking to try and do some recording and have something done by this weekend. I'll upload the .mp4 files to the [Doc Repository](https://github.com/GDquest/Blender-power-sequencer-docs/tree/master/tutorial_series)

**Comment by NathanLovato** [comment]
His subway tut is nice indeed. Thanks much for your help!
Note: let's avoid uploading mp4s to the repository, it'll reach full capacity too soon. It's best if you can put it on the cloud or on some other file storage website.

**Comment by NathanLovato** [comment]
I started shooting after work as cherry trees are reaching full bloom in Nara. Got limited hardware and experience but if you need it it'll make for some FOSS stock footage.

![image](https://user-images.githubusercontent.com/12694995/38121328-8e69dd58-3409-11e8-96ea-4e830e88164e.png)

Thinking of which, we could add a section about compositing/sequence modifiers

**Comment by doakey3** [comment]
Here's my [Video Editing Basics Tutorial](https://drive.google.com/file/d/1Ez2NRtcLQoe9L7W24lc9LQSmc-KRI1aS/view?usp=sharing)

**Comment by NathanLovato** [comment]
Jeez, you're handsome and you present well! How about you drop the medical field and work in FOSS/education instead? ;)
It's a great section for starters. Are you okay with the idea of making one sizeable video split into chapters, to get people started this time, rather than a playlist? I explained the reasons why I prefer it last time but if you'd rather have a playlist and individual videos, considering all the help you provide, we can go for that.

**Comment by doakey3** [comment]
[Here's a short one](https://drive.google.com/open?id=1ihYlJHd2XCmmAFz6yRGPfF0P-bEdPf-t) on making proxies and drawing audio waveforms

**Comment by NathanLovato** [comment]
Thanks much! When I'm done with the current Godot course chapter and article I'm working on, I'll take care of the first tut on my end. I'm dying to do some of this ^^

Great style again for the proxy one. Just a few notes because the way you explained proxies can be confusing:

1. You say the problem is Blender has to decode every frame of the base video. That's always the case, and the issue with most formats is that it has to decode 10 to 30 frames at once to rebuild one frame of the video. For continuous playback it's not a big deal because video players will pre-seek and read video streams in chunks but that's why it gets choppier when you jump to different points in time in a video editor (there's also some tiny lag if you do that in video readers, right).
2. Proxies are not un-encoded. They're compressed in such a way that it's easier for blender to read individual images: in this case blender converts your video in a series of JPG files - every frame becomes a self-contained chunk of data on the disk so it'll be as fast to jump to any frame on the timeline.
3. It'd be worth adding that 25% proxies
  1. Mean smaller resolution, so smaller pictures to decode, so faster playback than 50% or 100% proxies (and it's the biggest factor when it comes to performances)
  2. Are about the same resolution as your preview window in the top right so they're a perfect fit in your case

No need to re-record in this case: you can overlay a text note or a simple vector illustration (please tell me if you need me to do some graphic design work, no problem)

**Comment by doakey3** [comment]
Alright, let's not use it then. I'll see if I can fix it.

**Comment by NathanLovato** [comment]
Nah no worries, it's easy enough to patch. Adding a bit of text as an erratum on the video is fine. And rerecording a short segment is also good just these 10 seconds. If you're really concerned about the flow of the video fading to a slide, or fading you can out for a few seconds and overlaying a small illustration.
In general, if and when you need graphic design please ask.

**Comment by NathanLovato** [comment]
Hey daniel, I started recorded this morning after finishing another programming tutorial for work. Sorry it's taking time: it's a little complicated here with construction going on next door and living with 6 other persons. E.g. right now the cleaning lady's here, using the vacuum cleaner ^^
Is it okay for you if I do a take on the editing basics again but with the add-on?

**Comment by doakey3** [comment]
Yup, you can do what you feel needs to be done. 6 other people in the house would make it hard to record tutorials!

**Comment by NathanLovato** [comment]
Okay doing this to start with. It's mostly I don't want to overlap with your work or steal a topic from you.

> 6 other people in the house would make it hard to record tutorials!

Let's say it's an interesting exercise ;). Construction going on from 8 to 17-18, 6 to 7 days a week is more of a problem actually. It gets noisy randomly.

Took notes, it's calm now, gonna record the chunks. key-mon is linux only - i'm on Windows - and I don't know of any cross-platform tool to display keys. It's no big deal but just giving you a heads up: I'll be working with my usual keycastow. I've tried a few programs and none looks like key-mon. A pity 'cause it looks good.

**Comment by davcri** [comment]
Join the penguin side... *cough cough*

However great work guys! I'm following the development, hopefully I'll have
some free time to be more actively in the next weeks.

Sorry for the intrusion :D


Il mer 18 apr 2018, 07:03 Nathan Lovato <notifications@github.com> ha
scritto:

**Comment by NathanLovato** [comment]
Rendering my 3 little tuts now, gonna edit an intro and stitch all 5 parts together today. I'll post the result here as soon as it's done.

**Comment by NathanLovato** [comment]
Here's a 720p version of the tutorials! https://www.dropbox.com/s/12swqggyvf4ohle/05-29-get-started-with-video-editing-in-blender-with-power-sequencer-720p.mp4?dl=0

I made it smaller so you can download it quickly, as it's ~40mn. @doakey3 can you tell me if it's okay for you?

**Comment by doakey3** [comment]
Looks fine. I like how you used footage from [Crosscode](http://www.cross-code.com/en/home). I love that game.
